xarray/core/accessor_dt.py

```diff
@@ -358,31 +360,48 @@ def isocalendar(self):
 
         return Dataset(data_vars)
 
-    year = Properties._tslib_field_accessor(
-        "year", "The year of the datetime", np.int64
-    )
-    month = Properties._tslib_field_accessor(
-        "month", "The month as January=1, December=12", np.int64
-    )
-    day = Properties._tslib_field_accessor("day", "The days of the datetime", np.int64)
-    hour = Properties._tslib_field_accessor(
-        "hour", "The hours of the datetime", np.int64
-    )
-    minute = Properties._tslib_field_accessor(
-        "minute", "The minutes of the datetime", np.int64
-    )
-    second = Properties._tslib_field_accessor(
-        "second", "The seconds of the datetime", np.int64
-    )
-    microsecond = Properties._tslib_field_accessor(
-        "microsecond", "The microseconds of the datetime", np.int64
-    )
-    nanosecond = Properties._tslib_field_accessor(
-        "nanosecond", "The nanoseconds of the datetime", np.int64
-    )
+    @property
+    def year(self) -> T_DataArray:
+        """The year of the datetime"""
+        return self._date_field("year", np.int64)
+
+    @property
+    def month(self) -> T_DataArray:
+        """The month as January=1, December=12"""
+        return self._date_field("month", np.int64)
+
+    @property
+    def day(self) -> T_DataArray:
+        """The days of the datetime"""
+        return self._date_field("day", np.int64)
+
+    @property
+    def hour(self) -> T_DataArray:
+        """The hours of the datetime"""
+        return self._date_field("hour", np.int64)
```

Comment:
Thanks a lot for doing all these; this does seem like a nicer design!

Reply:
I couldn't figure out how to define the return types when assigning property types directly.
So that was the only solution.